chain-test/src/test/scala/org/bitcoins/chain/blockchain/sync/ChainSyncTest.scala

```diff
@@ -46,4 +46,30 @@ class ChainSyncTest extends ChainUnitTest {
 
       }
   }
+
+
+  it must "not fail when syncing a chain handler that is synced with it's external data source" in {
```

Comment:
This test text is a bit unclear to me, WYM?

Reply:
The intention here is that if we are in sync with our third party data source (we have the same blockchain tip) we shouldn't walk back on the chain.

This was actually covered before this commit, but there was an edge case where if our third party data source was _behind us on the blockchain_ we would still walk all the way back on our chain until we got to the genesis block. This was because we were only checking the tips, _not to see if the header existed already in the chain database_. If the header exists, we are making the assumption we have properly synced after that header.